src/Compilers/Test/Utilities/CSharp/FunctionPointerUtilities.cs

```diff
@@ -128,6 +130,42 @@ static void verifyParameter(ParameterSymbol symbol, MethodSymbol containing)
             }
         }
 
+        public static void VerifyPublicFunctionPointerGetTypeInfo(SemanticModel model, SyntaxNode syntax, bool useConvertedType,
+            Action<ITypeSymbol> returnTypeVerifier,
+            params Action<ITypeSymbol>[] paramVerifiers)
+        {
+            var typeInfo = model.GetTypeInfo(syntax);
+            var typeSymbol = (IFunctionPointerTypeSymbol)(useConvertedType ? typeInfo.ConvertedType : typeInfo.Type)!;
```

Comment:
>typeSymbol [](start = 16, length = 10)

Instead of taking a bunch of delegates, couldn't we just assert ```typeSymbol.ToTestDisplayString()```?

Reply:
The intent is to verify _both_ the full function pointer type, and the type we get from calling `GetTypeInfo` on the individual parts of the syntax. I felt this was best achieved by this pattern.

---
In reply to: [428922101](https://github.com/dotnet/roslyn/pull/44436#discussion_r428922101) [](ancestors = 428922101)